rfc/20241023-rfc-tracking-issues.md

```diff
@@ -0,0 +1,209 @@
+- "rfc-implementing": One or more specific individuals have signed up to implement what's described in the RFC and are actively working on it. At this stage the tracking issue should also be assigned to each of the assigned individuals.
+
+While the issue is in the "ready" or "implemented" phases it's possible that the folks working on it will learn of new blockers that were not originally anticipated, in which case they can make a judgement call about whether the blockers are significant enough to return to the "rfc-blocked" state or if they can be resolved promptly.
+
+The tracking issue conventionally "belongs to" some specific individual who is acting as a project owner for the implementation. In the short term that is likely to be a core team member, but in the long run could potentially be a community member who is willing and able to take on that role. That individual is ultimately responsible for keeping the main issue description and labels updated as the status changes, which implies that the individual must either be the author of the issue or must have "triage" access to the repository under [GitHub's access control scheme](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization). In practice the project owner may of course delegate some or all of these responsibilities to others; the requirement is only that there be a single person accountable for knowing the overall status of the work and able to communicate that with other stakeholders.
+
+Discussion about the implementation of the RFC should happen primarily in the comments on the tracking issue; the overall goal is that anyone who is interested in the project can find everything related to it either directly in the tracking issue or by following links from the tracking issue. Any linked feature request issues can continue to collect additional use-case examples and any other relevant information participants wish to share, without those becoming mixed in with the implementation-specific discussion in the RFC tracking issue.
+
+An RFC tracking issue is "closed as completed" (in GitHub's terms) once all of the code required for it has been completed and merged into the appropriate release branch. When closing the tracking issue the project owner, possibly after discussion with other stakeholders, should decide whether the completion of the RFC has completely satisfied each of the linked feature request issues and close them if so. It's okay for an RFC to only partially solve a problem, if there's a reasonably-clear path to fully solving the problem in future work, in which case the project owner should describe the situation in a comment on the original use-case issue and leave it open to represent the need for future work.
```

Comment:
In the situation I was intending to describe here, the "partial solution" would already be merged to `main`, and so could be built from there.

The kind of "partial solution" I'm thinking about here is, for example, if we decided to respond to https://github.com/opentofu/opentofu/issues/1329 by writing a separate RFC for each `lifecycle` argument, since all of them have different constraints and concerns and so it'd presumably be less risky to approach it gradually. If we took that approach then there would be several RFCs and several tracking issues all linking back to the same feature request, but the feature request _as a whole_ would be closed only once all of those RFC tracking issues are closed.

I think you are considering a slightly different situation where a single RFC gets implemented over multiple PRs and where in some situations it might not be appropriate to merge the partial work to `main` until the rest is completed. If so, I agree that it's important to make it clear exactly which branch the intermediate work is being merged into, and I agree with your earlier comment about ensuring that branch and/or PR are clearly linked from the RFC tracking issue, along with everything else.